### ISSUE TITLE
Fail to login using az devops login and a PAT

### ISSUE THREAD

**Issue by raj550** [Azure/azure-cli-extensions#1086]
### Extension name (the extension in question)
- az devops

### Description of issue (in as much detail as possible)
- Logging in using `az devops login` and a PAT is unsuccessful.
As a result, it fails in further steps (for example, a failure during `az artifacts universal download`)

Attempted installation via the following installation modes:
  - `az extension add devops`
  - `az extension add -y --source azure_devops-0.14.0-py2.py3-none-any.whl`

#### Debug logs for `az devops login`
```
Command arguments: ['devops', 'login', '--debug']
Event: Cli.PreExecute []
Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x7f240e91c1e0>, <function OutputProducer.on_global_arguments at 0x7f240e493d90>, <function CLIQuery.on_global_arguments at 0x7f240e234e18>]
Event: CommandInvoker.OnPreCommandTableCreate []
Installed command modules ['acr', 'acs', 'advisor', 'ams', 'apim', 'appconfig', 'appservice', 'backup', 'batch', 'batchai', 'billing', 'botservice', 'cdn', 'cloud', 'cognitiveservices', 'configure', 'consumption', 'container', 'cosmosdb', 'deploymentmanager', 'dla', 'dls', 'dms', 'eventgrid', 'eventhubs', 'extension', 'feedback', 'find', 'hdinsight', 'interactive', 'iot', 'iotcentral', 'keyvault', 'kusto', 'lab', 'managedservices', 'maps', 'monitor', 'natgateway', 'netappfiles', 'network', 'policyinsights', 'privatedns', 'profile', 'rdbms', 'redis', 'relay', 'reservations', 'resource', 'role', 'search', 'security', 'servicebus', 'servicefabric', 'signalr', 'sql', 'sqlvm', 'storage', 'vm']
Loaded module 'acr' in 0.006 seconds.
Loaded module 'acs' in 0.016 seconds.
Loaded module 'advisor' in 0.001 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'ams' in 0.007 seconds.
Loaded module 'apim' in 0.002 seconds.
Loaded module 'appconfig' in 0.003 seconds.
Loaded module 'appservice' in 0.009 seconds.
Loaded module 'backup' in 0.003 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'batch' in 0.008 seconds.
Loaded module 'batchai' in 0.003 seconds.
Loaded module 'billing' in 0.001 seconds.
Loaded module 'botservice' in 0.003 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'cdn' in 0.002 seconds.
Loaded module 'cloud' in 0.001 seconds.
Loaded module 'cognitiveservices' in 0.001 seconds.
Loaded module 'configure' in 0.001 seconds.
Loaded module 'consumption' in 0.002 seconds.
Loaded module 'container' in 0.002 seconds.
Loaded module 'cosmosdb' in 0.007 seconds.
Loaded module 'deploymentmanager' in 0.003 seconds.
Loaded module 'dla' in 0.004 seconds.
Loaded module 'dls' in 0.003 seconds.
Loaded module 'dms' in 0.002 seconds.
Loaded module 'eventgrid' in 0.002 seconds.
Loaded module 'eventhubs' in 0.003 seconds.
Loaded module 'extension' in 0.001 seconds.
Loaded module 'feedback' in 0.001 seconds.
Loaded module 'find' in 0.001 seconds.
Loaded module 'hdinsight' in 0.002 seconds.
Loaded module 'interactive' in 0.000 seconds.
Loaded module 'iot' in 0.004 seconds.
Loaded module 'iotcentral' in 0.001 seconds.
Loaded module 'keyvault' in 0.006 seconds.
Loaded module 'kusto' in 0.002 seconds.
Loaded module 'lab' in 0.003 seconds.
Loaded module 'managedservices' in 0.001 seconds.
Loaded module 'maps' in 0.002 seconds.
Loaded module 'monitor' in 0.005 seconds.
Loaded module 'natgateway' in 0.001 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'netappfiles' in 0.003 seconds.
Loaded module 'network' in 0.031 seconds.
Loaded module 'policyinsights' in 0.002 seconds.
Loaded module 'privatedns' in 0.004 seconds.
Loaded module 'profile' in 0.001 seconds.
Loaded module 'rdbms' in 0.005 seconds.
Loaded module 'redis' in 0.002 seconds.
Loaded module 'relay' in 0.003 seconds.
Loaded module 'reservations' in 0.001 seconds.
Loaded module 'resource' in 0.007 seconds.
Loaded module 'role' in 0.005 seconds.
Loaded module 'search' in 0.002 seconds.
Loaded module 'security' in 0.002 seconds.
Loaded module 'servicebus' in 0.005 seconds.
Loaded module 'servicefabric' in 0.002 seconds.
Loaded module 'signalr' in 0.001 seconds.
Loaded module 'sql' in 0.008 seconds.
Loaded module 'sqlvm' in 0.002 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'storage' in 0.031 seconds.
Loaded module 'vm' in 0.014 seconds.
Loaded all modules in 0.257 seconds. (note: there's always an overhead with the first module loaded)
Extensions directory: '/root/.azure/cliextensions'
Found 1 extensions: ['azure-devops']
Extensions directory: '/root/.azure/cliextensions'
Extension compatibility result: is_compatible=True cli_core_version=2.0.76 min_required=2.0.69 max_required=None
Extensions directory: '/root/.azure/cliextensions'
Loaded extension 'azure-devops' in 0.022 seconds.
Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x7f240e1bf1e0>]
az_command_data_logger : command args: devops login --debug
metadata file logging enabled - writing logs to '/root/.azure/commands'.
Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x7f240e1d7f28>]
Event: CommandInvoker.OnPostArgumentLoad []
Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x7f240e194a60>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x7f240e194b70>]
Event: CommandInvoker.OnCommandTableLoaded []
Event: CommandInvoker.OnPreParseArgs [<function _documentdb_deprecate at 0x7f240c03c1e0>]
Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x7f240e493e18>, <function CLIQuery.handle_query_parameter at 0x7f240e234ea0>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x7f240e194ae8>, <function handler at 0x7f240bf09b70>, <function DevCommandsLoader.post_parse_args at 0x7f240b715620>]
Extensions directory: '/root/.azure/cliextensions'
Extensions directory: '/root/.azure/cliextensions'
az_command_data_logger : extension name: azure-devops
az_command_data_logger : extension version: 0.14.0
No tty available.
Getting PAT token in non-interactive mode.
Event: CommandInvoker.OnTransformResult [<function _resource_group_transform at 0x7f240e18b488>, <function _x509_from_base64_to_hex_transform at 0x7f240e18b510>]
Event: CommandInvoker.OnFilterResult []
Event: Cli.PostExecute []
az_command_data_logger : exit code: 0
Suppress exception There are no active accounts.
Suppress exception Please run 'az login' to setup account.
telemetry.save : Save telemetry record of length 2372 in cache
telemetry.check : Negative: The /root/.azure/telemetry.txt was modified at 2019-11-11 12:04:11.980324, which in less than 600.000000 s
command ran in 0.347 seconds.
```

### Additional System information
**OS:** CentOS Linux 7.7.1908
#### Contents of `az --help`
```
azure-cli                         2.0.76

command-modules-nspkg              2.0.3
core                              2.0.76
nspkg                              3.0.4
telemetry                          1.0.4

Extensions:
azure-devops                      0.14.0
Python (Linux) 3.6.8 (default, Aug  7 2019, 17:28:10)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

-----

**Comment by atbagga** [comment]
@raj550 

The debug log suggests that the terminal is non-interactive. 
> No tty available.
   Getting PAT token in non-interactive mode.

You will either need to pipe in the PAT in the az devops login command or you can use environment variable to login by setting the environment variable (AZURE_DEVOPS_EXT_PAT) with PAT token.

In an interactive environment when you run az devops login you will be prompted for PAT token. 
Alternatively if you set the environment variable as above, you do not need to run the login command. Each command will pick the PAT from environment variable and proceed.

**Comment by atbagga** [comment]
Looks like you might be running into the same issue as this - https://github.com/Azure/azure-devops-cli-extension/issues/878

Please add the --org param to your login command to see if that helps.

**Comment by raj550** [comment]
@atbagga 

Adding `--org` to the login command did the trick.
Thanks.

**Comment by atbagga** [comment]
This is fixed in version 0.15.0 and will be released within this week.

**Comment by atbagga** [comment]
New version is released with the fix.
